From 031b8192efdf0183aae4101c90f63bede8251a34 Mon Sep 17 00:00:00 2001
From: Danny Baumann <dannybaumann@web.de>
Date: Fri, 8 Jun 2018 11:40:34 +0200
Subject: [PATCH 2/3] ext4_utils: Fix FS creation for filesystems with exactly
 32768 blocks.

In that case, total block count matches number of blocks per group,
which yields precisely 1 group. last_group_size in that case was
calculated as 0, which together with group count 1 incorrectly triggered
the file size check. Special case that scenario by assigning
last_group_size as 32768 in that case.

Change-Id: I3b6e236a4d6951aca732324b85e46b8c4b5757b7
---
 ext4_utils/ext4_utils.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ext4_utils/ext4_utils.c b/ext4_utils/ext4_utils.c
index cbfd350f..7be28537 100644
--- a/ext4_utils/ext4_utils.c
+++ b/ext4_utils/ext4_utils.c
@@ -123,7 +123,8 @@ void ext4_create_fs_aux_info()
 
 	aux_info.default_i_flags = EXT4_NOATIME_FL;
 
-	u32 last_group_size = aux_info.len_blocks % info.blocks_per_group;
+	u32 last_group_size = aux_info.len_blocks == info.blocks_per_group
+		? aux_info.len_blocks : aux_info.len_blocks % info.blocks_per_group;
 	u32 last_header_size = 2 + aux_info.inode_table_blocks;
 	if (ext4_bg_has_super_block((int)aux_info.groups - 1))
 		last_header_size += 1 + aux_info.bg_desc_blocks +
-- 
2.17.1

